index.ts

```diff
@@ -56,9 +56,12 @@ const plugin = ({
           conditionNames:
             build.initialOptions.conditions ??
             pick({
+              browser: isBrowser,
+              module: isBrowser && isImport,
               import: isImport,
               require: isRequire,
               node: isNode,
+              default: true,
             }),
```

Comment:
Not sure about `module: isBrowser && isImport`. 🤔

I know there are packages doing [strange things](https://github.com/vitejs/vite/blob/5396a706ca21b9943e1c94d87192eeb4f03d3df5/packages/vite/src/node/plugins/resolve.ts#L689-L697) with the top-level `browser` and `module` fields. But even then, using UMD might be sub-optimal in some cases, but still would be correct.

Anyway, it's for the top-level fields. I don't know `module` condition is used like that in the wild (haven't seen it), but I personally would expect it to be equivalent in meaning to the `module` top-level field, which is a legitimate Node.js entry point.

Have you encountered packages that use `module` condition specifically for targeting the browser?


And as for `default`, according to `enhanced-resove`'s [test cases](https://github.com/webpack/enhanced-resolve/blob/e5ff68aef5ab43b8197e864181eda3912957c526/test/exportsField.js#L126-L142) it works by default, so no need to specify it explicitly here.

Reply:
Went with just adding `browser` condition in the end.
Published the fix under `0.5.1` version to NPM.